### DISCUSSION TITLE
Assertion `nsIter != appIter->second.fdToDomain.end()' failed.

### DISCUSSION THREAD

**Discussion by Meraroo** [ANLAB-KAIST/KENSv3#35] [Project Q&A 1. Application Layer (Socket)]
kens v3.3.1
WSL2 Ubuntu 20.04 / macOS Monterey 12.2.1
gcc 9.3.0 / clang 13.0.0

I have sizeof-array-related error.
In the serverMain function, when I define like
char buf[20]; 
I passed all test cases (when size of array is less than 25)
![image](https://user-images.githubusercontent.com/81629557/159289021-6c8e7a44-1afa-4eda-b3e1-44da4fe0ee24.png)
![image](https://user-images.githubusercontent.com/81629557/159289052-371bdcc2-8813-45fa-a6fe-7f1e9733959e.png)

But, if size of buffer is greater than or equal to 25, I have assertion failed error message saying
Assertion `nsIter != appIter->second.fdToDomain.end()' failed.
![image](https://user-images.githubusercontent.com/81629557/159288915-95f42361-5812-441b-8e02-980aafad2c6d.png)
![image](https://user-images.githubusercontent.com/81629557/159289289-21563da9-57d8-46ae-972d-938654f8850e.png)
![image](https://user-images.githubusercontent.com/81629557/159289341-41f6a8eb-6016-4dca-b49e-ba73cb7a3f61.png)
Similar thing happens when I define array with new or malloc.

Is this a intended error? I totally don't know why these error happens..
*In my macOS condition, even buffer with size 16 makes error..

**Comment by Pusnow** [comment]
It might be a memory bug. Follow below instructions to detect which part causes the error.

https://github.com/ANLAB-KAIST/KENSv3/wiki/Tip:-Debugging-Memory-Bugs-(Only-for-macOS-and-Linux)
